managed/resource_user.go

```diff
@@ -0,0 +1,295 @@
+	if !readOK {
+		resp.Diagnostics.AddError("Unable to read the state of the user ", message)
+		return
+	}
+	tflog.Debug(ctx, "User Create: User on read from API server", map[string]interface{}{
+		"User": createdUser})
+
+	diags := resp.State.Set(ctx, &createdUser)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func getIDsFromUserState(ctx context.Context, state tfsdk.State, user *User) {
+	state.GetAttribute(ctx, path.Root("account_id"), &user.AccountID)
+	state.GetAttribute(ctx, path.Root("project_id"), &user.ProjectID)
+	state.GetAttribute(ctx, path.Root("email"), &user.Email)
+	state.GetAttribute(ctx, path.Root("role_name"), &user.RoleName)
+	state.GetAttribute(ctx, path.Root("user_id"), &user.UserID)
+}
+
+func resourceUserRead(accountId string, projectId string, email string, userId string, apiClient *openapiclient.APIClient) (user User, readOK bool, errorMessage string) {
+	userResp, response, err := apiClient.AccountApi.ListAccountUsers(context.Background(), accountId).Email(email).Execute()
+	if err != nil {
+		errMsg := getErrorMessage(response, err)
+		return user, false, errMsg
+	}
+
+	user.AccountID.Value = accountId
+	user.ProjectID.Value = projectId
+	user.UserID.Value = userId
+	user.Email.Value = email
+	user.RoleName.Value = userResp.Data[0].Info.GetRoleList()[0].GetRoles()[0].Info.GetDisplayName()
+	user.UserName.Value = userResp.Data[0].Spec.GetFirstName() + userResp.Data[0].Spec.GetLastName()
+	user.UserState.Value = string(userResp.Data[0].Info.State)
+
+	return user, true, ""
+}
+
+// Read user
+func (r resourceUser) Read(ctx context.Context, req tfsdk.ReadResourceRequest, resp *tfsdk.ReadResourceResponse) {
+	var state User
+	getIDsFromUserState(ctx, req.State, &state)
+
+	tflog.Debug(ctx, "User Read: User email and role from state", map[string]interface{}{
+		"Email":    state.Email,
+		"RoleName": state.RoleName})
+
+	user, readOK, message := resourceUserRead(state.AccountID.Value, state.ProjectID.Value, state.Email.Value, state.UserID.Value, r.p.client)
+	if !readOK {
+		resp.Diagnostics.AddError("Unable to read the state of the user ", message)
+		return
+	}
+	tflog.Debug(ctx, "User Read: User on read from API server", map[string]interface{}{
+		"User": user})
+
+	diags := resp.State.Set(ctx, &user)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update user
+func (r resourceUser) Update(ctx context.Context, req tfsdk.UpdateResourceRequest, resp *tfsdk.UpdateResourceResponse) {
+	var plan User
+	resp.Diagnostics.Append(getUserPlan(ctx, req.Plan, &plan)...)
+	if resp.Diagnostics.HasError() {
+		tflog.Debug(ctx, "Error while getting the plan for the user")
+		return
+	}
+
+	apiClient := r.p.client
+	var state User
+	getIDsFromUserState(ctx, req.State, &state)
+	accountId := state.AccountID.Value
+	projectId := state.ProjectID.Value
+	userId := state.UserID.Value
+	email := state.Email.Value
+
+	roleName := plan.RoleName.Value
```

Comment:
You should probably check here `if new_user != old_user` and then throw an error since you mentioned the username can't be updated.

Reply:
Since user_name is a read only field now terraform doesn't allow us to provide a value for it in the config file
Please let me know if I am misunderstanding your comment